opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -987,13 +987,11 @@ def record_exception(
         escaped: bool = False,
     ) -> None:
         """Records an exception as a span event."""
-        try:
-            stacktrace = traceback.format_exc()
-        except Exception:  # pylint: disable=broad-except
-            # workaround for python 3.4, format_exc can raise
-            # an AttributeError if the __context__ on
-            # an exception is None
-            stacktrace = "Exception occurred on stacktrace formatting"
+        stacktrace = "".join(
+            traceback.format_exception(
+                type(exception), exception, exception.__traceback__
```

Comment:
Looking at the docs for `format_exception()` seems like the error type is no longer needed since 3.5

- > Changed in version 3.5: The etype argument is ignored and inferred from the type of value.
- > Changed in version 3.10: The etype parameter has been renamed to exc and is now positional-only.

It's pretty confusing across all the python versions but, would it work to
- pass only `value` and `tb` as named arguments
- leave a TODO to move to passing just exc [once 3.10 is oldest supported](https://docs.python.org/3.11/library/traceback.html#traceback.print_exception:~:text=Since%20Python%203.10%2C%20instead%20of%20passing%20value%20and%20tb%2C%20an%20exception%20object%20can%20be%20passed%20as%20the%20first%20argument.%20If%20value%20and%20tb%20are%20provided%2C%20the%20first%20argument%20is%20ignored%20in%20order%20to%20provide%20backwards%20compatibility.)?

Reply:
Sounds good